tests/mocked/conftest.py

```diff
@@ -0,0 +1,248 @@
+import json
+import os
+import ssl
+from collections.abc import Generator
+from typing import Literal, cast
+from uuid import uuid4
+
+import pytest
+from integration.test_helpers import temp_env_vars
+from moto import mock_aws
+from mypy_boto3_s3 import S3Client
+from mypy_boto3_stepfunctions import SFNClient
+from nhs_aws_helpers import (
+    s3_client as _s3_client,
+)
+from nhs_aws_helpers import (
+    ssm_client as _ssm_client,
+)
+from nhs_aws_helpers import (
+    stepfunctions,
+)
+from pytest_httpserver import HTTPServer
+from trustme import CA
+
+
+@pytest.fixture(scope="module", autouse=True)
+def _mock_aws():
+    with mock_aws():
+        yield
+
+
+@pytest.fixture(name="s3_client")
+def s3_client(_mock_aws) -> S3Client:
+    return _s3_client()
+
+
+@pytest.fixture(scope="session")
+def ca() -> CA:
+    return CA()
+
+
+@pytest.fixture(scope="session")
+def httpserver_ssl_context(ca: CA):
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+    localhost_cert = ca.issue_cert("localhost")
+    localhost_cert.configure_cert(context)
+    return context
+
+
+@pytest.fixture(scope="module")
+def _httpclient_ssl_context(ca: CA):
+    with ca.cert_pem.tempfile() as ca_temp_path:
+        ssl.create_default_context(cafile=ca_temp_path)
+
+
+@pytest.fixture()
+def environment(
+    _mock_aws, httpserver: HTTPServer, ca: CA
+) -> Generator[str, None, None]:
+    environment = uuid4().hex
+    with temp_env_vars(
+        ENVIRONMENT=environment,
+        AWS_REGION="eu-west-2",
+        AWS_EXECUTION_ENV="AWS_Lambda_python3.8",
+        AWS_LAMBDA_FUNCTION_NAME="lambda_test",
+        AWS_LAMBDA_FUNCTION_MEMORY_SIZE="128",
+        AWS_LAMBDA_FUNCTION_VERSION="1",
+        CRUMB_SIZE="10",
+        CHUNK_SIZE="10",
+        MESH_URL=f"https://localhost:{httpserver.port}",
+        MESH_BUCKET=f"{environment}-mesh",
+        SEND_MESSAGE_STEP_FUNCTION_ARN=f"arn:aws:states:eu-west-2:123456789012:stateMachine:{environment}-send-message",
+        GET_MESSAGES_STEP_FUNCTION_ARN=f"arn:aws:states:eu-west-2:123456789012:stateMachine:{environment}-get-messages",
+        CA_CERT_CONFIG_KEY=f"/{environment}/mesh/MESH_CA_CERT",
+        CLIENT_CERT_CONFIG_KEY=f"/{environment}/mesh/MESH_CLIENT_CERT",
+        CLIENT_KEY_CONFIG_KEY=f"/{environment}/mesh/MESH_CLIENT_KEY",
+        SHARED_KEY_CONFIG_KEY=f"/{environment}/mesh/MESH_SHARED_KEY",
+        MAILBOXES_BASE_CONFIG_KEY=f"/{environment}/mesh/mailboxes",
+    ):
+        yield environment
+
+
+@pytest.fixture()
+def mesh_s3_bucket(s3_client: S3Client, environment: str) -> str:
```

Comment:
i like that mesh returns a Bucket object here, rather than string bucket name. if it's never used to get the bucket though it's fine.